src/runtime/hexagon_remote/halide_remote.cpp

```diff
@@ -215,6 +196,105 @@ int halide_hexagon_remote_power_hvx_on() {
     return 0;
 }
 
+int halide_hexagon_remote_power_hvx_on_mode(int mode) {
+    int set_mips = 0;
+    unsigned int mipsPerThread = 0;
+    unsigned int mipsTotal = 0;
+    int set_bus_bw = 0;
+    uint64_t bwBytePerSec = 0;
+    unsigned int busbwUsagePercentage = 0;
+    int set_latency = 0;
+    int latency = 0;
+    int result = 0;
+
+    // Only need to compute perf parameters if power_ref_count is zero
+    if (power_ref_count == 0) {
+        HAP_power_response_t power_info;
+        power_info.type = HAP_power_get_max_mips;
+        int retval = HAP_power_get(NULL, &power_info);
+        if (0 != retval) {
+            log_printf("HAP_power_get(HAP_power_get_max_mips) failed (%d)\n", retval);
+            return -1;
+        }
+        unsigned int max_mips = power_info.max_mips;
+
+        power_info.type = HAP_power_get_max_bus_bw;
+        retval = HAP_power_get(NULL, &power_info);
+        if (0 != retval) {
+            log_printf("HAP_power_get(HAP_power_get_max_bus_bw) failed (%d)\n", retval);
+            return -1;
+        }
+        uint64 max_bus_bw = power_info.max_bus_bw;
+
+        // Sanity check max values
+        // The above API under-reports the max bus bw. If we use it as
+        // reported, performance is bad. Experimentally, this only
+        // needs to be ~10x.
+        // Make sure max values are at least:
+        const unsigned int default_mips = 500;
+        const uint64       default_bw   = 1000000000ULL;
+        if (max_bus_bw < default_bw) {
+            if (max_bus_bw == 0) {
+                max_bus_bw = default_bw;
+            }
+            while (max_bus_bw < default_bw) {
+                max_bus_bw *= 10;
+            }
+        }
+        if (max_mips < default_mips) {
+            max_mips = default_mips;
+        }
+
+        switch (mode) {
+            case halide_hvx_power_low:
+                set_mips               = TRUE;
```

Comment:
How about lifting/moving the common params out of the switch? set_mips, set_bus_bw, set_latency, latency?

Or, just calling halide_hexagon_power_hvx_on_perf, like it was before (but instead of the internal function)?